### PR TITLE
Exercise 10

### PR DESCRIPTION
**Pull request by elParcero**
Getting file size using NumpySeqHandler class.

### CODE REVIEW

File: exercises/10-exercise.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        return np.load('{}_{}.npy'.format(self._name, index))
+
+
+    def get_file_list(self, datum_kwarg_gen):
+        #This method is optional. It is not needed for access, but for export.
+        return ['{name}_{index}.npy'.format(name=self._name, **kwargs['index']) #**kwargs
+                for kwargs in datum_kwarg_gen]
+
+
+    def get_file_size(self, datum_gen):
```
Comment (jrmlhermitte) [inline exercises/10-exercise.py]: in the spirit of the change we discussed before let's remove `get_file_size` from here and rather call `get_file_list` and figure out the file size from there (outside of the class)

File: exercises/10-exercise.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    resources.append(resource)
+    #resources.add(resource['id'])
+   
+sizes = []
+# initialize a file handler
+# then retrieve the size for that specific file
+for resource in resources:
+	fh = NumpySeqHandler(resource['resource_path'], resource['root'])
+	datum_gen = db.reg.datum_gen_given_resource(resource)
+	sizes.append(fh.get_file_size(datum_gen))
```
Comment (jrmlhermitte) [inline exercises/10-exercise.py]: so let's just try:
```py3
file_list = fh.get_file_list(datum_gen)
file_sizes = [os.path.getsize(filename) for filename in file_list]
sizes.append(sum(file_sizes))


```

File: exercises/10-exercise.py [hunk 3]
```diff
@@ -38,27 +38,13 @@ def __call__(self, index):
 
     def get_file_list(self, datum_kwarg_gen):
         #This method is optional. It is not needed for access, but for export.
-        return ['{name}_{index}.npy'.format(name=self._name, **kwargs['index']) #**kwargs
+        return ['{name}_{index}.npy'.format(name=self._name, index =kwargs['datum_kwargs']['index']) #**kwargs
```
Comment (jrmlhermitte) [inline exercises/10-exercise.py]: I think here `datum_kwarg_gen` should generate a list of `datum_kwargs`.

For example, from datums, they're obtained by:
```py
datum_kwarg_gen = [datum['datum_kwargs'] for datum in datum_gen]
```

File: exercises/10-exercise.py [hunk 4]
```diff
@@ -90,14 +76,15 @@ def get_file_size(self, datum_gen):
...
 # then retrieve the size for that specific file
 for resource in resources:
-	fh = NumpySeqHandler(resource['resource_path'], resource['root'])
-	datum_gen = db.reg.datum_gen_given_resource(resource)
-	sizes.append(fh.get_file_size(datum_gen))
-    
-print(sizes)
\ No newline at end of file
+    fh = NumpySeqHandler(resource['resource_path'], resource['root'])
+    datum_gen = db.reg.datum_gen_given_resource(resource)
```
Comment (jrmlhermitte) [inline exercises/10-exercise.py]: try:
```py
datum_kwargs_list = [datum['datum_kwargs'] for datum in datum_gen]
```

Comment (elParcero) [reply exercises/10-exercise.py]: It gives me an error, I'll push the updated code.
TypeError: format() argument after ** must be a mapping, not int

Comment (jrmlhermitte) [reply exercises/10-exercise.py]: in lines furhter above:
```index =kwargs['datum_kwargs']['index']```

change to:
```index =kwargs['index']```

**Pull request merged**